umich_api/api_utils.py

```diff
@@ -75,13 +82,17 @@ def api_call(self, api_call, client_scope, method="GET", payload=None):
         :returns Requests object with response from API (or an Exception)
         :raises Exception: No Access Token
         """
-        return self.scopes[client_scope]["api_call"](api_call, self.scopes[client_scope]["access_token"], method, payload)
+        this_scope = self.scopes.get(client_scope)
+        return this_scope.get("api_call")(api_call, self.get_access_token(this_scope.get("token_url"), client_scope), method, payload)
```

Comment:
you don't want to process if the token is None. a check is needed exclusively for that